app/src/main/java/com/konarsubhojit/synckro/providers/onedrive/OneDriveAuthManager.kt

```
@@ -28,24 +44,290 @@ class OneDriveAuthManager @Inject constructor(
+                }
+
+                override fun onError(exception: MsalException) {
+                    if (!cont.isActive) return
+                    Timber.e(exception, "OneDriveAuthManager.currentAccounts: error")
+                    cont.resume(emptyList())
+                }
+            })
+        }
+    }
 
     override suspend fun acquireAccessToken(account: Account): AuthResult<String> {
```

detekt (1.23.8)

[warning] The function acquireAccessToken appears to be too complex based on Cognitive Complexity (complexity: 33). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)